sched/irq/irq_dispatch.c

```diff
@@ -184,6 +185,14 @@ void irq_dispatch(int irq, FAR void *context)
   sched_note_irqhandler(irq, vector, false);
 #endif
 
+#ifdef CONFIG_DEBUG_MM
+  if ((g_running_tasks[this_cpu()]->flags & TCB_FLAG_MEM_CHECK) || \
+       (this_task()->flags & TCB_FLAG_MEM_CHECK))
+    {
+      kmm_checkcorruption();
```

Comment:
We also do the same check in the idle thread, is this to catch an issue as early as possible?

Reply:
Yes, you are right,

For example:
thread1 -> thread2 -> thread3 -> idle -> thread4 -> crash
overwrite

If just check in idle, then can't get the "killer" as early as possible.
With this method, then we can check the "killer" at each context_switch.

Reply:
@Ouss4 how about your thinking?

Reply:
@Ouss4 But this is useful for debugging, if all the thread set TCB_FLAG_MEM_CHECK to flags, then once someone overwrite the memblock, then we can catch it immediately.